mlx/backend/metal/device.cpp

```diff
@@ -55,6 +74,26 @@ MTL::Library* load_library(
     }
   }
 
+#ifdef SWIFTPM_BUNDLE
+  // try to load from a swiftpm resource bundle -- scan the available bundles to
+  // find one that contains the named bundle
+  {
+    MTL::Library* library =
+        try_load_bundle(device, NS::Bundle::mainBundle()->bundleURL());
+    if (library != nullptr) {
+      return library;
+    }
+    auto bundles = NS::Bundle::allBundles();
```

Comment:
Note: this method is actually uncallable -- filed 119891330 and have a local fix (where this code will be consumed).